### ISSUE TITLE
How to open the scenario in Greenfoot? 

### ISSUE THREAD

**Issue by LittleBenjiBoy** [lfuelling/soultraps#77]
I downloaded the master branch as a zip and I opened the project.greenfoot file in soultraps-master\src\main\resources with Greenfoot and it opened an empty scenario. Can you give instructions on how to open the game with Greenfoot?

Thanks in advance

**Comment by lfuelling** [comment]
Hi,

the short answer is: You can't. The greenfoot file is there only because the engine needs it when the game is run as a java program, it's not a proper project file.

Soultraps is a Maven project, you'll need to install [Maven](https://maven.apache.org/download.cgi) and use the `mvn clean package` command to create the `target/soultraps.jar` file which then can be executed using `java -jar target/name-of.jar`.

You can also use an IDE like [IntelliJ](https://intellj.com) or [Eclipse](https://www.eclipse.org/ide/) to edit the source code and run the program from within the IDE.

If you want more information on how/why I did this, please check out [my blog post](https://lerks.blog/p/making-games-with-greenfoot-without-greenfoot) about using Greenfoot without having to use the Greenfoot IDE :)

**Comment by lfuelling** [comment]
Here's the long answer on how to run this, using IntelliJ (I tested it on macOS, should also work on Linux and WIndows, although the screenshots will look a bit different):

1. Download the zip and extract (or clone the repo using git)
2. Download IntelliJ (there is either a free "community edition" or a 30 day trail for the "ultimate edition", community is fine for this) using the link above
3. Open IntelliJ and in the initial window select "Open"
4. In the file dialog that appears, select the soultraps **folder**  (where the `pom.xml` is located) and click "open"
5. Open the "File" menu and select "Project Structure": 
![The File Menu](https://user-images.githubusercontent.com/5144843/165801307-aac196b2-fd58-4ef2-92bc-6ee57df4092e.png)
7. In the structure window, select the SDK dropdown and click "Download SDK": 
![The SDK menu](https://user-images.githubusercontent.com/5144843/165801464-2f1027d3-6a29-41f0-b6f3-9833a68ad0a3.png)
8. Select Java Version 1.8 (I'm using the Bellsoft JDK, others should work too) and click download: 
![The download selector](https://user-images.githubusercontent.com/5144843/165802026-716f4916-e51b-4934-a61f-2de7673128a2.png)
9. Click "Apply" in the project structure window and close it
10. Click the "Add Configuration" button: 
![The 'add configuration' button](https://user-images.githubusercontent.com/5144843/165800714-ce8a28d5-d267-414c-ac5e-a6e89e51fa74.png)
11. In the new window, click the plus sign and select "Application": 
![The 'add' menu](https://user-images.githubusercontent.com/5144843/165800951-1e312ea4-678f-4eb6-bfa2-ba6a0008a77c.png)
12. Fill out the form like shown in this screenshot: 
![The application config](https://user-images.githubusercontent.com/5144843/165802401-52e16f74-b136-4537-919c-c34f10c417d0.png)
13. You can now start the game using the "play" button: 
![The "run" button](https://user-images.githubusercontent.com/5144843/165802839-edae4a9a-abec-4025-84a3-518a7c4a41fa.png)

**Comment by LittleBenjiBoy** [comment]
Hello!

Thanks so much for the detailed explanation on how to get this game running. After following all your steps, I was not able to get the game running. I would get this error: "soultraps-master\src\main\java\io\lerk\soultraps\levels\Level.java:15:19
java: package javafx.util does not exist". I had to add the JavaFX.jar files to the Project Structure in Libraries, but even then, while the game was running, I was getting too many errors, like: java.io.IOException: Problem reading font data. The game's fonts weren't able to load like the one showed in the .gif that you have in the readme. 

Anyway, I wanted to be able to run this game solely using Greenfoot because I need to code a game with Greenfoot for a school project and wanted to get coding ideas, but thanks for your response and information. I will be sure to check out your site and read your blog post :).

**Comment by lfuelling** [comment]
No problem :)

Depending on your operating system you might need to use another JDK (install or set it using the project structure window). 
Maybe try installing Oracle 1.8 (although their website is a nightmare), this should contain JavaFX for most platforms.

EDIT: I totally forgot that [there is a prebuilt jar](https://github.com/lfuelling/soultraps/releases/tag/v0.0.3) for v0.0.3.
If you just want to run the game and debugging is not important you can use that.

**Comment by LittleBenjiBoy** [comment]
Yeah lol, their website is a nightmare. I had to depend on google to help me with all these JavaFX issues. I'm really a newbie when it comes to programming. My operating system is Windows 10. 

When I would try to run the prebuilt jar for v0.0.3 through CMD, I would get this error (I think it's a JavaFX issue): 
![image](https://user-images.githubusercontent.com/43734617/165891735-dbc656b1-88c9-4d7b-a82f-7577ca3a37f9.png)

Troubleshooting info:
My versions of java are: 
![image](https://user-images.githubusercontent.com/43734617/165893034-be3e4a09-1bd9-48ca-b58a-a0f07c36e381.png)

JDK Version: 
![image](https://user-images.githubusercontent.com/43734617/165893092-1f773cac-040e-4a79-bdb9-697e5eb7c86d.png)

**Comment by lfuelling** [comment]
The `javac` version seems to be wrong, it should be something like `~1.8`. Try running `java -version` in the shell, I assume the Java version is not 1.8. Java >= 11 does not include JavaFX and thus causes the class not found error. 

This could be fixed by setting the `JAVA_HOME` environment variable to the path of the version of Java you want to use, but I don't know how to do that on Windows unfortunately 🙈  
It could also be that there is a GUI way to do this, and that changing `JAVA_HOME` is not the correct way to set the Java version on Windows.
